fury/stream/client.py

```diff
@@ -0,0 +1,412 @@
+            Python >=3.8 it's requirement to use SharedMemory.
+        whithout_iren_start : bool, optional
+            Set that to True if you can't initiate the vtkInteractor
+            instance.
+
+        """
+
+        self.showm = showm
+        self.iren = self.showm.iren
+        if use_raw_array:
+            self.circular_queue = ArrayCircularQueue(
+                max_size=max_queue_size, dimension=_CQUEUE.dimension
+            )
+        else:
+            self.circular_queue = SharedMemCircularQueue(
+                max_size=max_queue_size, dimension=_CQUEUE.dimension
+            )
+
+        self._id_timer = None
+        self._id_observer = None
+        self._interval_timer = None
+        self._whithout_iren_start = whithout_iren_start
+        self._started = False
+
+    def start(self, ms=3, use_asyncio=False):
+        """Start the stream interaction client.
+
+        Parameters
+        ----------
+        ms : float, optional
+            positive number greather than zero.
+        use_asyncio : bool, optional
+            If False then the interaction will be performed in a
+            separate thread.
+
+        """
+
+        use_asyncio = platform.system() == 'Windows' or use_asyncio
+        if ms <= 0:
+            raise ValueError('ms must be greater than zero')
+
+        if self._started:
+            self.stop()
+        if self._whithout_iren_start:
+            Interval = IntervalTimer \
+                    if use_asyncio else IntervalTimerThreading
+            self._interval_timer = Interval(
+                ms/1000,
+                interaction_callback,
+                *[
+                     self.circular_queue,
+                     self.showm,
+                     self.iren,
+                     True
+                 ]
+            )
+        else:
+            def callback(caller, event, *args, **kwargs):
+                interaction_callback(
+                    self.circular_queue, self.showm, self.iren, True)
+
+            self._id_observer = self.showm.iren.AddObserver(
+                "TimerEvent", callback)
+            self._id_timer = self.showm.iren.CreateRepeatingTimer(ms)
+
+        self._started = True
+
+    def stop(self):
+        """Stop the stream interaction client.
+        """
+        if not self._started:
+            return False
+
+        if self._id_timer is not None:
+            # self.showm.destroy_timer(self._id_timer)
+            self.showm.iren.DestroyTimer(self._id_timer)
+            self._id_timer = None
+
+        if self._id_observer is not None:
+            self.showm.iren.RemoveObserver(self._id_observer)
+            self._id_observer = None
+
+        if self._interval_timer is not None:
+            self._interval_timer.stop()
+            self._interval_timer = None
+
+        self._started = False
+
+    def cleanup(self):
+        """Relsese the shared memory resources if necessary.
+        """
+        self.circular_queue.cleanup()
```

Comment:
It looks like a lot `FuryStreamClient start stop and cleanup`. 

Maybe there is a way to simplify this. Need some thinking